horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/SchemaServiceNoRestTest.java

```diff
@@ -0,0 +1,639 @@
+import org.junit.jupiter.api.Disabled;
+
+import java.util.List;
+
+import static io.hyperfoil.tools.horreum.svc.BaseServiceNoRestTest.DEFAULT_USER;
+import static io.hyperfoil.tools.horreum.svc.BaseServiceNoRestTest.FOO_TEAM;
+import static io.hyperfoil.tools.horreum.svc.BaseServiceNoRestTest.FOO_TESTER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@QuarkusTest
+@QuarkusTestResource(PostgresResource.class)
+@TestProfile(HorreumTestProfile.class)
+@TestTransaction
+@TestSecurity(user = DEFAULT_USER, roles = {Roles.TESTER, Roles.VIEWER, FOO_TEAM, FOO_TESTER})
+class SchemaServiceNoRestTest extends BaseServiceNoRestTest {
+
+   @Inject
+   SchemaService schemaService;
+
+   @org.junit.jupiter.api.Test
+   void testCreateSchema() {
+      String schemaUri = "urn:dummy:schema";
+      Schema schema = createSampleSchema("Dummy schema", schemaUri, FOO_TEAM);
+
+      // create the schema
+      int id = schemaService.add(schema);
+      assertTrue(id > 0);
+
+      SchemaDAO savedSchema = SchemaDAO.findById(id);
+      assertEquals(schema.name, savedSchema.name);
+      assertEquals(schema.uri, savedSchema.uri);
+      assertEquals(schema.owner, savedSchema.owner);
+   }
+
+   @TestSecurity(user = DEFAULT_USER)
+   @org.junit.jupiter.api.Test
+   void testCreateSchemaForbidden() {
+      String schemaUri = "urn:dummy:schema";
+      Schema schema = createSampleSchema("Dummy schema", schemaUri, FOO_TEAM);
+
+      assertThrows(ForbiddenException.class, () -> schemaService.add(schema));
+   }
+
+   @TestSecurity()
+   @org.junit.jupiter.api.Test
+   void testCreateSchemaUnauthorized() {
+      String schemaUri = "urn:dummy:schema";
+      Schema schema = createSampleSchema("Dummy schema", schemaUri, FOO_TEAM);
+
+      assertThrows(UnauthorizedException.class, () -> schemaService.add(schema));
+   }
+
+   @TestSecurity(user = DEFAULT_USER, roles = {Roles.TESTER})
+   @org.junit.jupiter.api.Test
+   void testCreateSchemaWithNotExistingOwner() {
+      String schemaUri = "urn:dummy:schema";
+      Schema schema = createSampleSchema("Dummy schema", schemaUri, FOO_TEAM);
+
+      assertThrows(SQLGrammarException.class, () -> schemaService.add(schema));
+   }
+
+
+   @org.junit.jupiter.api.Test
+   void testCreateSchemaWithInvalidUri() {
+      String schemaUri = "dummy:schema";
+      Schema schema = createSampleSchema("Dummy schema", schemaUri, FOO_TEAM);
+
+      ServiceException thrown = assertThrows(ServiceException.class, () -> schemaService.add(schema));
+      assertEquals("Please use URI starting with one of these schemes: [urn, uri, http, https, ftp, file, jar]", thrown.getMessage());
```

Comment:
In relation to the previous comment about the Exception message, something like 

```suggestion
      assertEquals(Response.Status.BAD_REQUEST.getStatusCode();, thrown.getResponse().getStatus());
      assertEquals("Please use URI starting with one of these schemes: [urn, uri, http, https, ftp, file, jar]", thrown.getMessage());
```

Reply:
Added for all tests that were checking the `ServiceException`

Reply:
great, I can see the changes now, might have been looking at an old version of the files :/